### DISCUSSION TITLE
Why Should `BlobStorageService` Be Registered as Singleton

### DISCUSSION THREAD

**Discussion by markgmarkg** [Daniel-Krzyczkowski/Cars-Island-On-Azure#8] [General]
Hi Daniel,

Quick "lifetime" question...
I perfectly understand why `BlobServiceClient` should be a singleton - it's MS guidance.
`services.AddSingleton(factory => new BlobServiceClient(storageConfiguration.ConnectionString));`

I'm wondering what is a reason to register `BlobStorageService` as singleton.
`services.AddSingleton<IBlobStorageService, BlobStorageService>();`

Thank you,
mg

**Comment by Daniel-Krzyczkowski** [comment]
Hi Mark,

Thank you - good question.
There reason why I decided to register _BlobStorageService_ as singleton is the fact that it does not use any other dependencies than _BlobServiceClient_ which is also registered as singleton. As _BlobServiceClient_ manages connections internally (as explained by Microsoft) I did not see any reasons to register _BlobStorageService_ as transient or scoped. 
Please let me know if it make sense.

**Reply by markgmarkg** [reply]
Hi Daniel,

Thank you for such a quick response :)
I would choose non-`singleton` scope only for one reason - singletons require some kind of synchronization...
_Blob_ operations may be lengthy, although no state really managed in `BlobStorageService`

Regards,
Mark

**Reply by Daniel-Krzyczkowski** [reply]
Hi Mark,

This is interesting suggestion, thank you - I will be 100% honest with you. I did deep-dived into details but I assumed that if my class is using only this one singleton instance (_BlobServiceClient_) I assumed that there is no sense to register it in other way. I did not found any cons around this. I do not say that my solution is the only one and perfect. :)
